### PR TITLE
Feature/role model

### PR DESCRIPTION
**Pull request by apeksha2697**
**WHY**
We need to design skeleton of firm-form, where we'll have multiple organisations, each organisation containing multiple employees(user)

**CHANGES**
This feature contains 'organisation', 'role' and 'user' models.
1. added organisation model.
2. added unique index to name column for organisation model.
3. added presence validations to name column for organisation model.
4. created Role model.
5. added presence validations to name column for role model.
6. added not null to name column for role model.
7. added not null to name column for organisation model.
8. added uniqueness validations to name for organisation model.
9. added gem annotate to gemfile.
10. added User model.
11. added associations between user and organisation and user and role models.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#  twitter_url     :string
+#  linkedin_url    :string
+#  github_url      :string
+#  organisation_id :bigint           not null
+#  role_id         :bigint           not null
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#
+class User < ApplicationRecord
+  belongs_to :organisation
```
Comment (BandanaPandey) [inline app/models/user.rb:23]: @apeksha2697  Here, please separate associations and validations by one-line spacing, as well as segregate by adding comments like #associations and #validations

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  belongs_to :role
+  validates :first_name, presence: true
+  validates :email, presence: true, uniqueness: true
+  validates :gender, inclusion: { in: %w(male female other) }
```
Comment (BandanaPandey) [inline app/models/user.rb:29]: @apeksha2697  good work here, adding %w

File: test/models/organisation_test.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+#  id          :bigint           not null, primary key
+#  name        :string           not null
+#  description :text
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+require 'test_helper'
+
+class OrganisationTest < ActiveSupport::TestCase
+  # test "the truth" do
```
Comment (BandanaPandey) [inline test/models/organisation_test.rb]: @apeksha2697  whenever you get time please add testcases here, or if you need help let me know.
Whenever you'll add testcases, please fix the above fixtures data also

File: app/models/user.rb (hunk 1 above)
Comment (apeksha2697) [reply app/models/user.rb:23]: done

File: db/seeds.rb [hunk 4]
```diff
@@ -5,3 +5,14 @@
...
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+# Dir[File.join(Rails.root, 'db', 'seeds', '*.rb')].sort.each { |seed| load seed }
+
+$LOAD_PATH << '.'
+require 'db/seeds/organisation_seeds.rb'
+require 'db/seeds/role_seeds.rb'
+require 'db/seeds/user_seeds.rb'
+
+SeedOrganisation.create_organisation
```
Comment (BandanaPandey) [inline db/seeds.rb]: @apeksha2697  name, should be like this
`Organisation.seed_data`
`Role.seed_data`
`User.seed_data`

Comment (apeksha2697) [reply db/seeds.rb]: done

**Pull request merged**